File: problemas/matematica/1630/1630.cs
using System;
using System.Linq;
using System.Collections.Generic;

class URI {
    static int mdc(int a, int b){
        return (b == 0) ? a : mdc(b, a % b);
    }

    static void Main(string[] args) {
        int X, Y;

        while(true){
            try{
                List<int> entrada = Console.ReadLine().Trim().Split(' ').Select((x) => int.Parse(x)).ToList();

                X = entrada[0];
                Y = entrada[1];

                Console.WriteLine(2 * (X/mdc(X, Y) + Y/mdc(X, Y)));
            }catch (Exception ex){
                break;
            }
        }
    }
}

File: problemas/matematica/1630/1630.swift
import Foundation

func MDC(a: Int, b: Int) -> Int {
    if(b == 0) {
        return a;
    }
    return MDC(a: b, b: a % b);
}

while let linha = readLine(strippingNewline: true) {
    let entrada = linha.components(separatedBy: " ")
    let x = Int(entrada[0])!
    let y = Int(entrada[1])!

    print(2 * (x / MDC(a: x, b: y) + y / MDC(a: x, b: y)))
}
